### PR TITLE
OSS to Enterprise migration overhaul

### PR DESCRIPTION
**Pull request by sanderson**

### CODE REVIEW

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 1]
```diff
@@ -4,46 +4,40 @@ aliases:
...
-* Requires downtime for writes and reads for the OSS instance
+- Deletes all data from any data nodes that are already part of an InfluxDB Enterprise cluster
+- Transfers all users from the OSS instance to the InfluxDB Enterprise Cluster
+- Requires downtime for writes and reads for the OSS instance
 
-{{% warn %}}
-If you're using an InfluxDB Enterprise cluster version prior to 0.7.4, the
-following steps will **not** transfer users from the OSS instance to the
-InfluxDB Enterprise Cluster.
+In addition, please refrain from creating a Global Admin user in the InfluxDB Enterprise Web Console before implementing these steps.
```
Comment (danatinflux) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: Do we even have Global Admin users any longer? I thought the web console went away before 1.6?

Comment (sanderson) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: Good point. I'll take that out 😄.

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 2]
```diff
@@ -1,235 +1,285 @@
 ---
 title: Migrate InfluxDB OSS instances to InfluxDB Enterprise clusters
+description: >
+  Migrate a running instance of InfluxDB open source (OSS) and to an InfluxDB Enterprise Cluster.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: extra "and" > (OSS) to

global update > cluster

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 3]
```diff
@@ -1,235 +1,285 @@
...
-## IMPORTANT
-Due to a known issue in InfluxDB, attempts to upgrade an InfluxDB OSS instance to
-InfluxDB Enterprise will fail.
-A fix is in place and will be released with InfluxDB v1.7.10.
-Until InfluxDB v1.7.10 is released, **DO NOT** attempt to migrate an InfluxDB OSS
-instance to InfluxDB Enterprise by following the steps in this guide.
-
-We will update this guide to reflect the new upgrade process after the release of InfluxDB 1.7.10.
-{{% /warn %}}
+Migrate a running instance of InfluxDB open source (OSS) and to an InfluxDB Enterprise Cluster.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: ditto

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 4]
```diff
@@ -1,235 +1,285 @@
...
 
-<!-- The following guide has step-by-step instructions for migrating an InfluxDB open source (OSS)
-instance into an InfluxDB Enterprise cluster.
+## Requirements
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: Prerequisites

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 5]
```diff
@@ -1,235 +1,285 @@
...
+- An InfluxDB OSS instance running **InfluxDB 1.7.10 or newer**.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: global update > or later

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 6]
```diff
@@ -1,235 +1,285 @@
...
+- An InfluxDB Enterprise cluster
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: does version of Enterprise matter?

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 7]
```diff
@@ -1,235 +1,285 @@
...
+- Network accessibility between the OSS instances and all data and meta nodes.
 
 {{% warn %}}
-The process assumes that you already have a fully configured InfluxDB Enterprise cluster
-of three or more meta nodes and zero or more data nodes. If you need instructions for meta node installation:
-- [Production installation of meta nodes](/enterprise_influxdb/v1.7/production_installation/meta_node_installation/)
+**Please note that this migration process:**
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: Migrating does the following:

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 8]
```diff
@@ -1,235 +1,285 @@
...
+
+- Deletes all data from any data nodes that are already part of an InfluxDB Enterprise cluster
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: Deletes existing data in InfluxDB Enterprise data nodes

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 9]
```diff
@@ -1,235 +1,285 @@
...
+- Transfers all users from the OSS instance to the InfluxDB Enterprise Cluster
+- Requires downtime for writes and reads for the OSS instance
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: for OSS instance reads and writes?

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 10]
```diff
@@ -1,235 +1,285 @@
...
+
+In addition, please refrain from creating a Global Admin user in the InfluxDB Enterprise Web Console before implementing these steps.
+If you’ve already created a Global Admin user, contact support.
 {{% /warn %}}
 
-Please note that this migration process:
+## Process overview
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: ## To migrate to InfluxDB Enterprise
Complete the following tasks:

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 11]
```diff
@@ -1,235 +1,285 @@
...
-{{% warn %}}
-If you're using an InfluxDB Enterprise cluster version prior to 0.7.4, the
-following steps will **not** transfer users from the OSS instance to the
-InfluxDB Enterprise Cluster.
-{{% /warn %}}
+## Upgrade to InfluxDB 1.7.10 or newer
 
-In addition, please refrain from creating a Global Admin user in the InfluxDB Enterprise Web Console before implementing these steps. If you’ve already created a Global Admin user, contact support.
+InfluxDB 1.7.10 (both OSS and Enterprise) fixed an issue that prevented upgrading
+and InfluxDB OSS instance to InfluxDB Enterprise successfully.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: and > an

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 12]
```diff
@@ -1,235 +1,285 @@
...
 
-## Modify the `/etc/hosts` file
+For upgrade instructions, see the following:
 
-Add the IP and hostname of the InfluxDB OSS instance to the
-`/etc/hosts` file on all nodes in the InfluxDB Enterprise cluster.
-Ensure that all cluster IPs and hostnames are also in the OSS
-instance’s `/etc/hosts` file.
+- [Upgrade InfluxDB OSS](/influxdb/latest/administration/upgrading/)
+- [Upgrade InfluxDB Enterprise](/enterprise_influxdb/v1.7/administration/upgrading/)
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: Upgrade to InfluxDB 1.7.10 or later:
- [Upgrade InfluxDB OSS](...)
- [Upgrade InfluxDB Enterprise](...)

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 13]
```diff
@@ -1,235 +1,285 @@
...
 
-**Note:** All node hostnames must be completely resolvable by all
-other nodes in the cluster. If you have DNS already setup in your
-environment, then this step can be skipped.
+## Set up InfluxDB Enterprise meta nodes
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: is this part of the InfluxDB Enterprise upgrade?

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 14]
```diff
@@ -1,235 +1,285 @@
...
 ```
+    {{% /code-tab-content %}}
+    {{< /code-tabs-wrapper >}}
+
+4. **Back up your InfluxDB OSS configuration file**
 
-### 4. Update the binary
+    If you have any custom configuration settings in your InfluxDB OSS configuration file,
+    copy that configuration file and keep it as a backup.
+    **After updating the InfluxDB binary, the configuration file stored at the default location will be overwritten**.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: If you have custom configuration settings for InfluxDB OSS, back up and save your configuration file. **Without a backup, you'll lose custom configuration settings when Updating the InfluxDB binary.**

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 15]
```diff
@@ -1,235 +1,285 @@
...
 
-> **Note:** This step will overwrite your current configuration file.
-If you have settings that you’d like to keep, please make a copy of your config file before running the following command.
+5. **Update the InfluxDB binary**
 
-#### Ubuntu and Debian (64-bit)
+    > This step will overwrite your current configuration file.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: Updating the InfluxDB binary overwrites the configuration file...

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 16]
```diff
@@ -1,235 +1,285 @@
...
+    If you have settings that you’d like to keep, please make a copy of your config file before running the following command.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: keep, back up your configuration file...

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 17]
```diff
@@ -1,235 +1,285 @@
...
+6. **Update the configuration file**
 
-In `/etc/influxdb/influxdb.conf`, set:
+    In `/etc/influxdb/influxdb.conf`, set:
 
-* `hostname` to the full hostname of the data node
-* `license-key` in the `[enterprise]` section to the license key you received on InfluxPortal **OR** `license-path`
-in the `[enterprise]` section to the local path to the JSON license file you received from InfluxData.
+    - `hostname` to the full hostname of the data node
+    - `license-key` in the `[enterprise]` section to the license key you received on InfluxPortal **OR** `license-path`
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: is InfluxPortal official name?

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 18]
```diff
@@ -1,235 +1,285 @@
...
+        {{% /warn %}}
 
-```toml
-# Hostname advertised by this host for remote addresses.  This must be resolvable by all
-# other nodes in the cluster
-hostname="<data-node-hostname>" #✨
+    <!-- -->
+    ```toml
+    # Hostname advertised by this host for remote addresses.
+    # This must be resolvable by all other nodes in the cluster.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: accessible?

File: content/enterprise_influxdb/v1.7/guides/migration.md (hunk 6 above)
Comment (sanderson) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: Yes, it does. I'll add it.

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 19]
```diff
@@ -1,235 +1,285 @@
...
-  license-key = "<your_license_key>" #✨ mutually exclusive with license-path
+    [enterprise]
+      # license-key and license-path are mutually exclusive,
+      # use only one and leave the other blank
+      license-key = "<your_license_key>"
 
-  # license-key and license-path are mutually exclusive, use only one and leave the other blank
-  license-path = "/path/to/readable/JSON.license.file" #✨ mutually exclusive with license-key
-```
+      # license-key and license-path are mutually exclusive,
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: think we can say once?

File: content/enterprise_influxdb/v1.7/guides/migration.md (hunk 9 above)
Comment (sanderson) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: I just changed this to "requires downtime..."

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 20]
```diff
@@ -1,235 +1,285 @@
...
 ```bash
 sudo systemctl start influxdb
 ```
+    {{% /code-tab-content %}}
+    {{< /code-tabs-wrapper >}}
 
-### 7. Add the node to the cluster
+
+## Add the new data node to the cluster
+With the OSS instance is upgrading to InfluxDB Enterprise, add it to the Enterprise cluster.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: is OSS a new data node? After you upgrade your OSS instance (node?) to Enterprise, add the node to your Enterprise cluster?

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 21]
```diff
@@ -1,235 +1,285 @@
...
-You can do this with [ALTER RETENTION POLICY](/influxdb/v1.7/query_language/database_management/#modify-retention-policies-with-alter-retention-policy).
-
-Next, [rebalance](/enterprise_influxdb/v1.7/guides/rebalance/) your cluster manually to meet the desired
-replication factor for existing shards.
-
-Finally, if you were using [Chronograf](/chronograf/latest/), you can
-add your Enterprise instance as a new data source.  If you were not using
-[Chronograf](/chronograf/latest/introduction/installation/), we recommend going through
-the installation instructions and using it as your primary management UI for the instance. -->
+Once added to the cluster, InfluxDB will begin synchronized data stored on the
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: begins to synchronize data on the upgraded...

File: content/enterprise_influxdb/v1.7/guides/migration.md (hunk 13 above)
Comment (sanderson) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: No, it's not.

File: content/enterprise_influxdb/v1.7/guides/migration.md (hunk 17 above)
Comment (sanderson) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: Yes, I believe it is. This is where Enterprise users manage license keys.

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 22]
```diff
@@ -1,235 +1,285 @@
...
+upgraded OSS node to other data nodes in the cluster.
+It may take a few minutes before the existing data is available.
+
+## Rebalance the cluster
+1. Use the [ALTER RETENTION POLICY](/influxdb/v1.7/query_language/database_management/#modify-retention-policies-with-alter-retention-policy)
+   statement to increase the [replication factor](/enterprise_influxdb/v1.7/concepts/glossary/#replication-factor)
+   on all existing retention polices to the number of data nodes in your cluster.
+2. [Rebalance your cluster manually](/enterprise_influxdb/v1.7/guides/rebalance/)
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: to update the specified replication factor for existing shards?

File: content/enterprise_influxdb/v1.7/guides/migration.md [hunk 23]
```diff
@@ -1,235 +1,285 @@
...
+   to meet the desired replication factor for existing shards.
+3. If you were using [Chronograf](/chronograf/latest/),add your Enterprise instance as a new data source.
```
Comment (kelseiv) [inline content/enterprise_influxdb/v1.7/guides/migration.md]: If you're using Chronograf,<space?> add...

File: content/enterprise_influxdb/v1.7/guides/migration.md (hunk 22 above)
Comment (sanderson) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: We don't specify what replication factor they should use. We give them guidelines to follow (covered in the rebalancing doc linked here), but what rf they go with is a matter of preference.

Comment (kelseiv) [reply content/enterprise_influxdb/v1.7/guides/migration.md]: Gotcha. My goal here was to drop "desired" and clarify what "meet" means. 
Would this be accurate: [Rebalance your cluster manually} to update the replication factor for existing shards" 

or is there a better word than "update"?

**Pull request merged**